doc/neps/nep-0036-fair-play.rst

```diff
@@ -80,6 +80,10 @@ Fair play rules
    include the phrase `numpy`, i.e., avoid names such as
    `mycompany_numpy`.
 
+   To be clear, this rule only applies to modules (package names); it
+   is perfectly acceptable to have a *submodule* of your own library
+   named `mylibrary.numpy`.
```

Comment:
A general comment for this NEP: This `` ` `` and many of the others should be ``` `` ```

Reply:
`mycompany.numpy` vs `my_company.numpy` also feels like a somewhat arbitrary distinction to me. In particular, I'm wondering if my package https://github.com/eric-wieser/ros_numpy (written before I contributed much to numpy) falls afoul of these rules, and if so what name would have been preferred by this NEP.

Reply:
to add another point: there are more than a few packages on pypi with numpy in the name when searching [site:pypi.org/ numpy](https://duckduckgo.com/?q=site%3Apypi.org%2F+numpy&ia=web)

Reply:
I guess that's the whole point: to discourage people from naming their packages like that without due consideration.

If the project is closely related to NumPy, a community project, and aptly named, all it takes is getting in touch with the team and asking them if it's fine by them.  I wouldn't consider `ros_numpy` a problem, e.g..

Reply:
Also, we're not going to police this; these are guidelines, and we specify them so that we can point to them in the future if they are contravened.

Reply:
> `mycompany.numpy` vs `my_company.numpy` also feels like a somewhat arbitrary distinction to me. In particular, I'm wondering if my package https://github.com/eric-wieser/ros_numpy (written before I contributed much to numpy) falls afoul of these rules, and if so what name would have been preferred by this NEP. Naming it `ros.numpy` is only possible if you're in control of the `ros` namespace, which I'm not.

I don't understand what you mean; both `mycompany.numpy` and `my_company.numpy` are considered fine.  However, `mycompany_numpy` is not, because `numpy` appears as part of the package name.

Reply:
Sorry, that was an unfortunate typo.

Reply:
Also, we should maybe clarify the distinction between package name (the entry in PyPI and setup.py), and module name (the entry in sys.modules).

Reply:
> If the project is closely related to NumPy, a community project, and aptly named, all it takes is getting in touch with the team and asking them if it's fine by them.

The language is confusing. Ont the one hand the sentence above this says "When in doubt, please talk to us first.", but then this one says "Do not reuse the NumPy name ..." with no qualifier. That doesn't sound like it invites the conversation in the introduction.

Reply:
It is very confusing and not really good practice to have a package name that is different from the module name.

Reply:
> A general comment for this NEP: This `` ` `` and many of the others should be `` `` ``,

I pushed a fix for this.

> `mycompany.numpy` vs `mycompany_numpy` also feels like a somewhat arbitrary distinction to me.

A submodule is very different from a package name (e.g. doesn't show up in a PyPI search). For submodules the `numpy` name can make perfect sense, e.g. `jax.numpy` is the compat module for JAX that provides a numpy-like API.